src/stores/WalletStore.ts

```diff
@@ -42,19 +41,26 @@ class WalletStore extends PersistableStore {
     return web3Modal.cachedProvider
   }
 
-  async connect(clearCachedProvider = false) {
+  async connect(clearCachedProvider = false, checkNetwork = false) {
     this.walletLoading = true
     try {
       if (clearCachedProvider) web3Modal.clearCachedProvider()
 
       const instance = await web3Modal.connect()
       provider = new Web3Provider(instance)
-      const userNetwork = (await provider.getNetwork()).name
-      await this.checkNetwork(provider, userNetwork)
-      if (this.needNetworkChange)
+      if (
+        checkNetwork &&
+        !(await this.checkNetwork()) &&
```

Comment:
![image](https://user-images.githubusercontent.com/49658988/199033827-341f6651-d5c4-429a-ad32-44c5eb38d415.png)
Those look similar, we need to extract it into `checkNetowrkAndRequestChange` function, I think